### PR TITLE
Importers: Move action buttons to the bottom of each screen

### PR DESCRIPTION
**Pull request by spen**
This PR addresses issues with buttons in both the [importers that expect the user to upload a file](https://github.com/Automattic/wp-calypso/issues/31092) and the [ones that expect them to input a URL](https://github.com/Automattic/wp-calypso/issues/30564), namely:

* Inconsistencies in button positioning.
* Unclear button text.
* Clashing button colours.

At the moment, buttons appear in different places in the importer pane, depending on the type of importer and stage of the import. @dezzie's Issues #31092 and #30564 propose a more consistent placement of all buttons in the bottom left corner of the importer pane. They also recommend some small changes to button text, and suggest removing the "scary" style from some buttons whose colour seems to clash with the magenta of the new Muriel primary style buttons.

#### Testing
Run through the whole importing flow with both a Wix import and a file-based import (WordPress).
Try cancelling at various points in the flow - does the button do what you expected in each place? Does the copy best represent the action?
Take the the flows through to completion - did the loading states look right? Did you get stuck at any point?

#### Note
The approach used here isn't perfect - because of limitations around state and certain actions being baked in to components (rather than abstracted in to the state layer) I've had to place the `ActionButton`s wherever they have access to to the actions they need to trigger: In the `AuthorMappingPane`, `SiteImporterInputPane `,  `SiteImporterSitePreview `, `ImportingPane` and `UploadingPane` components.
The alternative would be to properly move out various pieces of state and state-changing actions from the components and in to redux but that has proven tricky in the past and wouldn't be worthwhile at this point.

#### Suggestion for extra work:

Between entering a URL & hitting enter, and the site-preview being generated there are essentially 2 loading states: The continue button shows it's 'busy' state (while we check the URL's validity), then there's just a spinner (while we generate the mShot preview).
It would be a smoother experience if we could have the busy state of the button represent both the validation and the mShot call.

These screenshots show how the UI of the file upload importers would look with this PR applied.

**Import start**

image

**Import file uploading**
(Ignore the excessive width of the progress bar in these screenshots - we'll fix that in a separate PR.)

image

**Import file uploaded**

image

**Author mapping**

image

**Import started**

image

**Importing resources progress**

image

**Import done**

image

### CODE REVIEW

**Comment by andfinally** [comment]
The code looks good to me - I did run into a weird state issue locally with my Wix import though, so I'd like to give this some more testing tomorrow. I'm not sure if the issue was related to these changes or if it was a pre-existing glitch.

**WP import**

* `Start import` opens importer
* `Cancel` closes importer
* `Cancel` is greyed out during upload
* `Cancel` after successful upload clears uploaded file from file pane
* Upload pane successfully uploads new file after Chrome file upload error interrupts big upload
* `Continue` after upload goes to author mapping view
* `Cancel` leaves author mapping view and returns to importers list
* `Start Import` starts import - button changes to `Importing` with busy state
* `Done` after import finished returns to importers list.

**Wix Import**

* `Start import` opens importer, `Continue` is greyed out
* `Cancel` closes importer
* `Continue` starts processing
* `Cancel` during processing returns to importers list
* `Cancel` on preview pane returns to URL input pane
* `Yes! Start import` on preview pane starts import
* `Importing` with busy state appears (shortly after spinner) during import
* `Cancel` on importer failure goes to importers list

**Wix weirdness**

* My Wix import got stuck. It seemed to be taking a long time. When I eventually refreshed the page, I got the importers list with all importers greyed out - the `ImporterStatus` for all of them was `importer-disabled`. 
* On refreshing again I got just the "stuck" view of the active Wix importer again with the status `importer-importing`.
* I reset the importer state in my blog RC and started a new import with a different URL. 
* I got through the preview pane and started the import. 
* The view jumped back to the URL input pane.
* Eventually it changed again to the empty importer with the cancel button. (Status was `importer-import-failure`.) This was the kind of error state I came across earlier today.

File: client/my-sites/importer/author-mapping-pane.jsx [hunk 1]
```diff
@@ -11,11 +11,14 @@ import React from 'react';
...
  */
-import Button from 'components/forms/form-button';
 import AuthorMapping from './author-mapping-item';
 import SiteUsersFetcher from 'components/site-users-fetcher';
 import UsersStore from 'lib/users/store';
 
+import ActionButtonContainer from 'my-sites/importer/importer-action-buttons/container';
+import ActionButton from 'my-sites/importer/importer-action-buttons/action-button';
+import CloseButton from 'my-sites/importer/importer-action-buttons/close-button';
+
```
Comment (creativecoder) [inline client/my-sites/importer/author-mapping-pane.jsx]: These components seem odd to me, in that they have generic names but are importing specific. Could we have more descriptive names, like `ImportActionContainer`, `ImportActionButton`, `ImportCancelButton`, etc?

(I know this issue isn't new to this PR)

File: client/my-sites/importer/importer-action-buttons/close-button.jsx [hunk 2]
```diff
@@ -59,15 +59,9 @@ export class CloseButton extends React.PureComponent {
...
-				className="importer-header__action-button"
-				disabled={ disabled }
-				isPrimary
-				scary
-				onClick={ this.handleClick }
-			>
-				{ translate( 'Close', { context: 'verb, to Close a dialog' } ) }
-			</Button>
+			<ActionButton disabled={ disabled } onClick={ this.handleClick }>
+				{ translate( 'Cancel', { context: 'verb, to Close a dialog' } ) }
```
[the remark this answers is not in the document]
Comment (creativecoder) [reply client/my-sites/importer/importer-action-buttons/close-button.jsx]: +1

**Comment by creativecoder** [comment]
Pinging @dezzie for a design review.

**Comment by andfinally** [comment]
I don't know if it's just me, but it seems like one effect of the change in the upload pane is to make it less apparent when a file has successfully uploaded - previously, a magenta button popped into view in the middle of the pane, making it very clear the upload had completed. I think the upload success is less obvious now.

image

image

image

**Comment by andfinally** [comment]
I've done a bunch of imports - several Wix, an Importer 6, a Medium - and all went well, so I reckon we're good to go. I'll hold off approving for now in case Spen wants to push any changes.

**Comment by spen** [comment]
> The code looks good to me - I did run into a weird state issue locally with my Wix import though, so I'd like to give this some more testing tomorrow. I'm not sure if the issue was related to these changes or if it was a pre-existing glitch.

Sadly, this is all existing weirdness, the only remedy would be to properly address the weird state set up we have there :/

> I don't know if it's just me, but it seems like one effect of the change in the upload pane is to make it less apparent when a file has successfully uploaded...

I see what you're saying. I wonder if a nice ✅ icon or similar would be a good indicator here?
Another improvement I can see would be to have the continue button present throughout the upload process, but have it disabled until the file is uploaded, similar to how it is in the URL input pane before it has a value.

File: client/my-sites/importer/uploading-pane.jsx [hunk 3]
```diff
@@ -121,6 +118,8 @@ class UploadingPane extends React.PureComponent {
 	};
 
 	render() {
+		const { importerStatus, site, isEnabled } = this.props;
+		const { importerState, importerId } = this.props.importerStatus;
```
[the remark this answers is not in the document]
Comment (spen) [reply client/my-sites/importer/uploading-pane.jsx]: Good spotting on these! Copy pasta I swear!
Those should be handled in https://github.com/Automattic/wp-calypso/pull/31364/commits/7617ce80ec9c53d0697c04abb17f8dcf17245276 :)

**Comment by dezzie** [comment]
@spen, @andfinally, @creativecoder, yay! I took a look, and ran some imports, and I think we are good to merge. I hear you on:
- Having something that indicates a successful file upload has taken place in the upload area. The text message is better than nothing, so I would go with what we have now if doing something new would hold us up a few days.
- @spen's note about the suggested extra work to smooth out the progress indicators.

If we can follow-up on those things and put them in a separate PR for efficiency that'd be great. That also gives me time to run this stuff by the Muriel team without holding stuff up.

**Comment by spen** [comment]
> I'll hold off approving for now in case Spen wants to push any changes.

All planned changes are up - the component naming and destructuring fixes, so ready for another 👀!
Thanks a bunch for the thorough reviews @andfinally :)

**Comment by spen** [comment]
> I've created #31481 for the file upload success UI.

Awesome, thanks @andfinally :) And thanks for testing again! 🙏

**Comment by bsessions85** [comment]
@spen  It looks like this change broke an e2e test. Can you take a shot at updating the `Import a site while signing up` test? I would guess it is probably just a change in a css selector

**Comment by spen** [comment]
Thanks for the heads up @bsessions85 - I'll get on this right away :)

**Pull request merged**